service/operator/service.go

```diff
@@ -1,40 +1,118 @@
 package operator
 
 import (
+	"fmt"
 	"sync"
 
+	"k8s.io/client-go/kubernetes"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/util/intstr"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/giantswarm/kvmtpr"
 	microerror "github.com/giantswarm/microkit/error"
 	micrologger "github.com/giantswarm/microkit/logger"
+	"github.com/giantswarm/operatorkit/client/k8s"
+	"github.com/giantswarm/operatorkit/tpr"
+)
+
+const (
+	PortNameFormat = "ingress-controller-port-%s"
 )
 
 // Config represents the configuration used to create a new service.
 type Config struct {
 	// Dependencies.
-	Logger micrologger.Logger
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	// Settings.
+	Namespace string
+	Service   string
 }
 
 // DefaultConfig provides a default configuration to create a new service by
 // best effort.
 func DefaultConfig() Config {
+	var err error
+
+	var k8sClient kubernetes.Interface
+	{
+		config := k8s.DefaultConfig()
+		k8sClient, err = k8s.NewClient(config)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	var newLogger micrologger.Logger
+	{
+		config := micrologger.DefaultConfig()
+		newLogger, err = micrologger.New(config)
+		if err != nil {
+			panic(err)
+		}
+	}
+
 	return Config{
 		// Dependencies.
-		Logger: nil,
+		K8sClient: k8sClient,
+		Logger:    newLogger,
+
+		// Settings.
+		Namespace: "default",
+		Service:   "ingress-controller",
 	}
 }
 
 // New creates a new configured service.
 func New(config Config) (*Service, error) {
 	// Dependencies.
+	if config.K8sClient == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "config.K8sClient must not be empty")
```

Comment:
I find this more descriptive and helpful for the developer hustling around the project.